packages/core/src/index.ts

```diff
@@ -173,138 +174,126 @@ export const initialize = <T extends Record<string, any> = Record<string, any>>(
-      .map(async (scriptLocation: string) => {
-        const data = await injectScript(scope, scriptLocation);
-        resolvePendingInjection(scope);
-        return data;
-      })
-  );
-  setPendingInjection(scope, injectionPromise);
-  injectionPromise.then((res) => {
-    resolvePendingInjection(scope);
-    return res;
-  });
-  return injectionPromise;
-}
-
-export async function asyncLoader<T = any, P = any>(scope: string, module: string): Promise<ExposedScalprumModule<T, P>> {
-  if (typeof scope === 'undefined' || scope.length === 0) {
-    throw new Error("Scope can't be undefined or empty");
-  }
-  if (typeof module === 'undefined' || module.length === 0) {
-    throw new Error("Module can't be undefined or empty");
-  }
-
-  if (!module.startsWith('./')) {
-    console.warn(`Your module ${module} doesn't start with './' this indicates an error`);
-  }
-
-  await __webpack_init_sharing__('default');
-  const container: Container = (window as { [key: string]: any })[scope];
-  await container.init(__webpack_share_scopes__.default);
-  const factory = await (window as { [key: string]: any })[scope].get(module);
-
-  if (!window[GLOBAL_NAMESPACE].factories[scope]) {
-    window[GLOBAL_NAMESPACE].factories[scope] = {} as any;
+    await pendingInjection;
+    const exposedModule = await pluginStore.getExposedModule<ExposedScalprumModule>(scope, module);
+    setExposedModule(getModuleIdentifier(scope, module), exposedModule);
+    return;
   }
 
-  const moduleFactory = factory();
-  const factoryCache: Factory = {
-    init: container.init,
-    modules: {
-      ...window[GLOBAL_NAMESPACE].factories[scope].modules,
-      [module]: {
-        ...moduleFactory,
-      },
-    },
-    expiration: new Date(),
-  };
+  // eslint-disable-next-line no-async-promise-executor
+  pendingInjection = new Promise<void>(async (res, rej) => {
+    const headers = new Headers();
+    headers.append('Pragma', 'no-cache');
+    headers.append('Cache-Control', 'no-cache');
+    headers.append('expires', '0');
+    const manifestPromise = fetch(url, {
+      method: 'GET',
+      headers,
+    });
+    const response = await manifestPromise;
+
+    // handle network errors
+    if (!response.ok) {
+      let error = 'Unable to process manifest';
+      try {
+        error = await response.json();
+      } catch {
+        // try text error if json fails
+        error = await response.text();
+      }
+
+      return rej(error);
+    }
+    // response is OK get manifest payload
+    const manifest = await response.json();
+    const loadScripts: string[] = processor ? processor(manifest) : manifest[scope].entry;
+
+    // TODO: Add option to change base URL
+    const injectionScript = pluginStore.loadPlugin(document.location.origin, {
+      extensions: [],
+      loadScripts,
+      name: scope,
+      registrationMethod: 'custom',
+      version: '1.0.0',
```

Comment:
Does this mean that ConsoleDot / HCC plugin manifest currently doesn't provide version information?

Reply:
Currently, they don't provide that info.